checks/runchecks.go

```diff
@@ -29,14 +29,21 @@ func RunChecks(ctx *context.Context) []*pkg.CheckResult {
 
 func transformResults(ctx *context.Context, in []*pkg.CheckResult) (out []*pkg.CheckResult) {
 	for _, r := range in {
+		// If check fails, do not apply transform
+		if !r.Pass {
```

Comment:
!pass doesn't necessarily means it failed to connect - it could have failed a threshold, response code/body check etc.

Reply:
I went through the checks code and all the errors are raised via `.Failf` or `.ErrorMessage` (which calls `.Failf` inside) all of which sets pass = false.

If not for r.Pass, what should we check against ?

Reply:
We can add another field called Error = true that ErrorMessage sets - however we might need to update some of the checks to use ErrorMessage instead of Failf for connection errors